Makefile

```diff
@@ -1,125 +1,152 @@
-	@$(RUN_TASK) install_cluster_deps "$(VENV)"
-	@$(RUN_TASK) kind_cluster_up "$(VENV)"
+.PHONY: all venv install install-dev build clean printvars help
+.PHONY: test test-unit test-functional test-e2e lint
+.PHONY: push push-dev push-rc push-prod push-rebuild
+.PHONY: dist-deb dist-rpm dist-mac
 
-.PHONY: cluster-down
-cluster-down: venv ## delete kind testing k8s cluster
-	@$(RUN_TASK) install_cluster_deps "$(VENV)"
-	@$(RUN_TASK) kind_cluster_down "$(VENV)"
+all: VERBOSE := true ## Run Anchore CLI full CI pipeline locally (lint, build, test, push)
+all: lint build test push
 
-.PHONY: push push-dev
-push: push-dev ## push dev image to dockerhub
-push-dev: 
-	@$(RUN_TASK) push_dev_image "$(COMMIT_SHA)" "$(DEV_IMAGE_REPO)" "$(GIT_BRANCH)" "$(TEST_IMAGE_NAME)"
+anchore-ci: ## Fetch test artifacts for local CI
+	rm -rf /tmp/test-infra; git clone $(TEST_HARNESS_REPO) /tmp/test-infra
+	mv ./anchore-ci ./anchore-ci-`date +%F-%H-%M-%S`; mv /tmp/test-infra/anchore-ci .
 
-.PHONY: push-rc
-push-rc: 
-	@$(RUN_TASK) push_rc_image "$(DEV_IMAGE_REPO)" "$(GIT_TAG)" "$(TEST_IMAGE_NAME)"
+venv: $(VENV)/bin/activate ## Set up a virtual environment
+$(VENV)/bin/activate:
+	python3 -m venv $(VENV)
 
-.PHONY: push-prod
-push-prod: 
-	@$(RUN_TASK) push_prod_image_release "$(DEV_IMAGE_REPO)" "$(GIT_BRANCH)" "$(GIT_TAG)"
+install: venv setup.py requirements.txt ## Install to virtual environment
+	@$(ACTIVATE_VENV) && $(PYTHON) setup.py install
 
-.PHONY: push-rebuild
-push-rebuild: 
-	@$(RUN_TASK) push_prod_image_rebuild "$(COMMIT_SHA)" "$(DEV_IMAGE_REPO)" "$(GIT_TAG)"
+install-dev: venv setup.py requirements.txt ## Install to virtual environment in editable mode
+	@$(ACTIVATE_VENV) && $(PYTHON) -m pip install --editable .
 
-.PHONY: venv
-venv: $(VENV)/bin/activate ## setup virtual environment
-$(VENV)/bin/activate:
-	@$(RUN_TASK) setup_venv "$(VENV)"
+lint: venv anchore-ci ## Lint code (currently using flake8)
+	@$(ACTIVATE_VENV) && $(CI_CMD) lint
+
+# Local CI script
+build: Dockerfile anchore-ci venv ## Build dev Anchore CLI Docker image
+	@$(CI_CMD) scripts/ci/build "$(COMMIT_SHA)" "$(GIT_REPO)" "$(TEST_IMAGE_NAME)"
+
+test: ## Run all tests: unit, functional, and e2e
+	@$(MAKE) test-unit
+	@$(MAKE) test-functional
+	@$(MAKE) test-e2e
+
+test-unit: anchore-ci venv ## Run unit tests (tox)
+	@$(ACTIVATE_VENV) && $(CI_CMD) test-unit
 
-.PHONY: install
-install: venv setup.py requirements.txt ## install project to virtual environment
-	@$(RUN_TASK) install "$(GIT_REPO)" "$(PYTHON)" "$(VENV)"
+test-functional: anchore-ci venv ## Run functional tests (tox)
+	@$(ACTIVATE_VENV) && $(CI_CMD) test-functional
 
-.PHONY: install-dev
-install-dev: venv setup.py requirements.txt ## install project to virtual environment in editable mode
-	@$(RUN_TASK) install_dev "$(GIT_REPO)" "$(PYTHON)" "$(VENV)"
+# Local CI scripts (set-e2e-tests and e2e-tests)
+test-e2e: anchore-ci venv ## Set up and run end to end tests
```

Comment:
I think this target should be broken down into multiple targets so I can run `make cluster-up` and get a running kind cluster, or `make run-e2e-tests` to rerun tests on an already running cluster.

Reply:
Agreed - I'm doing that for e2e and also integration and functional tests where it makes sense.